src/commands/mode.cpp

```diff
@@ -1,40 +1,356 @@
+            std::string message =
+                "missing argument for mode " + std::string(1, c);
+            return message;
+          }
+          ++argumentIndex;
+        }
+      }
+    }
+  }
+
+  return "";  // Return an empty string if no errors
+}
+
+std::string Server::checkModeString(const stringVector &modestringToCheck) {
+#ifdef DEBUG
+  {
+    std::ostringstream oss;
+    oss << "modestringToCheck.size(): " << modestringToCheck.size();
+    printLog(DEBUG_LOG, COMMAND, oss.str());
+  }
+#endif
+  if (modestringToCheck.empty()) {
+    return "";  // Return an empty string if modestringToCheck is empty
+  }
+
+  const std::string validModes = "itkol";
+  const std::string plusMinus = "+-";
+  for (size_t i = 0; i < modestringToCheck.size(); ++i) {
+    const std::string &mode = modestringToCheck[i];
+#ifdef DEBUG
+    {
+      std::ostringstream oss;
+      oss << "mode: " << mode;
+      printLog(DEBUG_LOG, COMMAND, oss.str());
+    }
+#endif
+    for (size_t j = 1; j < mode.size(); ++j) {
+      if (validModes.find(mode[j]) == std::string::npos) {
+#ifdef DEBUG
+        {
+          std::ostringstream oss;
+          oss << "Mode invalide: " << mode[j];
+          printLog(DEBUG_LOG, COMMAND, oss.str());
+        }
+#endif
+        return mode;
+      }
+    }
+  }
+  return "";
+}
+
+bool Server::isChannelValid(int fd, const std::string &channel) {
+  if (channel.empty() || channel[0] != REG_CHAN) {
+    send461NeedMoreParams(_clients[fd], "MODE");
+    return (false);
+  }
+  if (_channels.find(channel.substr(1)) == _channels.end()) {
+    send403NoSuchChannel(_clients[fd], channel);
+    return (false);
+  }
+  return (true);
+}
 
 void Server::mode(int fd, const std::string &arg) {
-/*--- To remove ---*/
-  (void)fd;
-  (void)arg;
-/*-----------------*/
+  const Client &client = _clients.at(fd);
+  std::istringstream iss(arg);
+  std::string channel, modeString;
+  iss >> channel;
+  if (isChannelValid(fd, channel) == false) return;
+  const Channel &channelObj = _channels[channel.substr(1)];
+
+  if (channelObj.isOperator(fd) == false) {
+    send482ChanOPrivsNeeded(client, channelObj);
+    return;
+  }
+  std::string remainingArgs;
+  std::getline(iss, remainingArgs);
+  KeyValuePairList modestringAndArguments = parseMode(remainingArgs);
+  if (modestringAndArguments.first.empty()) {
+    send324Channelmodeis(client, channelObj);
+    return;
+  }
+  stringVector modestringVector = modestringAndArguments.first;
+  stringVector modeArgumentsVector = modestringAndArguments.second;
+
+  std::string errorModeString = checkModeString(modestringVector);
+
+  if (!errorModeString.empty()) {
+    send472UnknownMode(client, errorModeString);
+    sendNotice(client, "usage: MODE <channel> {[+|-]|o|i|t|k|l} [<arguments>]");
+    return;
+  }
+  std::string errorModeArguments =
+      checkArguments(modestringVector, modeArgumentsVector);
+  if (!errorModeArguments.empty()) {
+    send461NeedMoreParams(client, errorModeArguments);
```

Comment:
461 reply format should be:
` <client> <command> :Not enough parameters`
-->  `:server 461 <nickname> MODE :Not enough parameters.`
 We cannot put a custom message for the numeric reply.